message_object.py

```diff
@@ -116,6 +116,10 @@ def __init__(self, **kwargs):
     self.x_velocity = kwargs.get("x_velocity")
     self.y_velocity = kwargs.get("y_velocity")
 
+    # Data from the Nav system with our target velocity. (m/s)
+    self.x_target_velocity = kwargs.get("x_target_velocity")
```

Comment:
Otherwise, it has no way of knowing when the velocity in the filter should change because the drone has changed directions. It needs *some* extra information. Either the target velocity, or the waypoints themselves, preferably the former.